src/lib/build_machine.ml

```diff
@@ -20,37 +19,22 @@ let create
     | None -> default_run_program ~host
     | Some r -> r
   in
-  let actual_b37 =
-    match b37 with
-    | None  ->
-      let destination_path = meta_playground // "B37-reference-genome" in
-      Data_providers.pull_b37 ~host ~run_program ~destination_path
-    | Some s -> s
-  in
+  let toolkit =
+    Tool_providers.default_toolkit ()
+      ~host ~meta_playground
+      ~gatk_jar_location ~mutect_jar_location in
   Machine.create (sprintf "ssh-box-%s" uri)
     ~ssh_name:(
       Uri.of_string uri |> Uri.host |> Option.value ~default:"No-NAME")
-    ~get_reference_genome:(function
-      | `B37 -> actual_b37
-      | `B38 -> 
-        Data_providers.pull_b38 
-          ~host ~run_program ~destination_path:(meta_playground // "B38-reference-genome")
-      | `hg18 ->
-        Data_providers.pull_hg18
-          ~host ~run_program ~destination_path:(meta_playground // "hg18-reference-genome")
-      | `hg19 -> 
-        Data_providers.pull_hg19 
-          ~host ~run_program ~destination_path:(meta_playground // "hg19-reference-genome")
-      | `B37decoy -> Data_providers.pull_b37decoy
-                       ~host ~run_program ~destination_path:(meta_playground // "hs37d5-reference-genome")
-      | `mm10 -> Data_providers.pull_mm10
-                   ~host ~run_program ~destination_path:(meta_playground // "mm10-reference-genome")
-      )
+    ~get_reference_genome:(fun name ->
+        match name, b37 with
+        | name, Some some37 when name = Reference_genome.name some37 -> some37
+        | name, _ ->        
+          Download_reference_genomes.get_reference_genome name
+            ~toolkit ~host ~run_program
+            ~destination_path:(meta_playground // "reference-genome"))
```

Comment:
The function `of_specification` actually creates a subdirectory with the genome name:
https://github.com/hammerlab/biokepi/pull/118/files#diff-443f45965cd1f3ecd1803f25ca33c469R22

And I've tested that :)
https://github.com/hammerlab/biokepi/pull/118/files#diff-cd61c399df66872fa5b8361429ff9c49R46

```
 $ ls biokepi-test-all-downloads/
b37  b37decoy  b38  hg18  hg19  mm10  tools
```